app/src/main/kotlin/deliveryfee/App.kt

```diff
@@ -0,0 +1,179 @@
+/*
+ * This Kotlin source file was generated by the Gradle 'init' task.
+ */
+package deliveryfee
+
+import io.ktor.application.*
+import io.ktor.features.ContentNegotiation
+import io.ktor.http.ContentType
+import io.ktor.request.receive
+import io.ktor.response.respond
+import io.ktor.routing.post
+import io.ktor.routing.routing
+import io.ktor.serialization.json
+import io.ktor.server.engine.embeddedServer
+import io.ktor.server.netty.Netty
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.Json
+import java.time.ZoneOffset
+import io.ktor.http.HttpStatusCode
+import java.time.DayOfWeek
+import java.time.OffsetDateTime
+
+
+@Serializable
+data class DeliveryRequest(
+    val cart_value: Int, // cent
+    val delivery_distance: Int, //meter
+    val number_of_items: Int,
+    val time: String // UTC
```

Comment:
all right, thanks!

Reply:
@dutch3883 how can I do that? 🧠
I thought this line is ensuring the format..

val deliveryTime = OffsetDateTime.parse(request.time)

Reply:
You can try using OffsetDateTime type directly. Otherwise, when you use  `OffsetDateTime.parse(request.time)` you can add try catch to handle the case of the program cannot parse `request.time` Failure

Reply:
@dutch3883 thanks. yeah, I tried to define the type directory, but it seems its incompatible with @Serializer cuz I couldn't address the error...

So I'm trying to adopt the second way tho, isn't my current code already have ability to detect the type error with the "try&catch" method?

Reply:
You can try call the api with invalid time string. It shouldn't response with server error but other code for example bad request: datetime format is incorrect.

Reply:
Alright, will try fix it: thanks!